tensorpipe/transport/ibv/reactor.cc

```diff
@@ -17,8 +17,32 @@ namespace ibv {
 
 Reactor::Reactor(IbvLib ibvLib, IbvDeviceList deviceList)
     : ibvLib_(std::move(ibvLib)) {
+  bool found = false;
   TP_DCHECK_GE(deviceList.size(), 1);
-  ctx_ = createIbvContext(getIbvLib(), deviceList[0]);
+
+  // If the deviceList contains multiple ibv devices, we will select the
+  // device of the port whose port_state is active, instead of just selecting
+  // the first device in the deviceList by default.
+  for (int i = 0; i < deviceList.size(); i++) {
+    IbvContext tp_ctx_;
```

Comment:
Our naming convention is `camelCase`. Also, a trailing underscore means that a name is a private class member, whereas this is just a local variable. Could you just name this `ctx`?